scripts/data/generate_data.py

```diff
@@ -265,24 +265,49 @@ def generate_data(
             block = fetch_block_header(next_block_hash)
         elif mode == "full":
             block = fetch_block(next_block_hash)
+
             for tx in block["data"]:
                 for idx, output in enumerate(tx["outputs"]):
                     outpoint = (tx["txid"], idx)
                     utxo_set[outpoint] = output
 
+                    output["cached"] = False
+
+            for tx in block["data"]:
                 for tx_input in tx["inputs"]:
                     outpoint = (
                         tx_input["previous_output"]["txid"],
                         tx_input["previous_output"]["vout"],
                     )
+
                     if outpoint in utxo_set:
+                        tx_input["previous_output"]["cached"] = True
                         utxo_set[outpoint]["cached"] = True
 
         else:
             raise NotImplementedError(mode)
         next_block_hash = block["nextblockhash"]
         blocks.append(block)
 
+    for block in blocks:
+        for tx in block["data"]:
+            for idx, output in enumerate(tx["outputs"]):
+                outpoint = (tx["txid"], idx)
+                if outpoint in utxo_set:
+                    tx["outputs"][idx]["cached"] = utxo_set[outpoint].get(
+                        "cached", False
+                    )
+
+            for tx_input in tx["inputs"]:
```

Comment:
You already did that in L284

Reply:
@Gerson2102 but you still need to iterate over all outputs (after all the blocks are applied) to set `cache=True` if they have `utxo_set[outpoint]["cached"]`

To sum up, you need to do two passes:
1. Build utxo set, update inputs which consume cached outputs
2. Update outputs that are marked as "cached" in the utxo set

Reply:
@m-kus check my last commit. This terms of outputs, outpoint, which I have to update is kinda confusing for me.